app/controllers/api/libraries_controller.rb

```diff
@@ -0,0 +1,39 @@
+class Api::LibrariesController < Api::BaseController
+  rescue_from Exceptions::PublishingError, with: :handle_publishing_error
+
+  def publish
+    Library.publish_components_and_styles(
+      components_params,
+      styles_params,
+      description_param,
+      designer_id_param
+    )
+    render_response({ message: 'Components and styles were published successfully.' })
+  end
+
+  private
+
```

Comment:
The `components_params` and `styles_params` methods should include more comprehensive validation to ensure that the input data is safe and valid. Consider adding additional checks or using a strong parameters approach.